testhelpers/docker_registry.go

```diff
@@ -64,143 +59,82 @@ func NewDockerRegistry(ops ...RegistryOption) *DockerRegistry {
 	return registry
 }
 
-func (r *DockerRegistry) Start(t *testing.T) {
-	t.Helper()
-
-	r.Host = DockerHostname(t)
-
-	t.Logf("run registry on %s", r.Host)
+func BasicAuth(handler http.Handler, username, password, realm string) http.Handler {
```

Comment:
Do this and `ReadOnly` need to be public?

Reply:
I was on the fence but since there's no auth built-into `ggcr/registry`, it could be useful to expose these. Also because this `imgutil` helper will be external to `lifecycle` and would need to be changed and bump to add functionality, leaving these public could make it easier to make a workaround while a change is being contributed into `imgutil`.

That is just hypothetical though, so if we'd rather make them private, that's fine with me.

Reply:
Makes sense! I'm happy to leave it as is.